agent/api/container/container_test.go

```diff
@@ -1334,6 +1336,73 @@ func TestGetCredentialSpec(t *testing.T) {
 	}
 }
 
+func TestRestartPolicy(t *testing.T) {
+	testCases := []struct {
+		name            string
+		container       *Container
+		restartCount    int
+		expectedEnabled bool
+	}{
+		{
+			name: "nil restart policy",
+			container: &Container{
+				RestartPolicy: nil,
+			},
+			restartCount:    0,
+			expectedEnabled: false,
+		},
+		{
+			name: "not enabled restart policy",
+			container: &Container{
+				RestartPolicy: &restart.RestartPolicy{},
+			},
+			restartCount:    0,
+			expectedEnabled: false,
+		},
+		{
+			name: "explicitly not enabled restart policy",
+			container: &Container{
+				RestartPolicy: &restart.RestartPolicy{
+					Enabled: false,
+				},
+			},
+			restartCount:    0,
+			expectedEnabled: false,
+		},
+		{
+			name: "enabled restart policy",
+			container: &Container{
+				RestartPolicy: &restart.RestartPolicy{
```

Comment:
Just curious, do we/are there plans to validate RestartPolicy isn't empty if restart policy is enabled when we receive the task payload?

Reply:
That's a good question, we may want to validate that when we initialize the restart trackers, let me sync with Fargate dataplane on how we want to do that

Reply:
👍 thanks Cam! I also have no issues for this to be a follow up